src/main/java/org/apache/commons/codec/digest/DigestUtils.java

```diff
@@ -99,6 +101,33 @@
         return updateDigest(messageDigest, data).digest();
     }
 
+    /**
+     * Reads through a File using non-blocking-io (NIO) and returns the digest for the data
+     *
+     * @param messageDigest The MessageDigest to use (e.g. MD5)
+     * @param data          Data to digest
+     * @return the digest
+     * @throws IOException On error reading from the stream
+     * @since 1.14
+     */
+    public static byte[] nonblockingDigest(final MessageDigest messageDigest, final File data) throws IOException {
```

Comment:
Don't use implementation details in method names, so -1 to the "nonblocking" prefix.

Reply:
If I remove the prefix from:

```java
nonblockingDigest(final MessageDigest messageDigest, final File data)
```

will clash with an existing method name.